internal/providers/provider.go

```diff
@@ -14,6 +14,10 @@ import (
 // Interface represents the set of methods required for a complete resource
 // provider plugin.
 type Interface interface {
+	// GetMetadata returns the initial information about the provider without
+	// requiring the provider to fully load schema information.
+	GetMetadata() GetMetadataResponse
```

Comment:
oh, since we're not using this in core right now, I think we can leave probably this out entirely. The interface is only there for internal polymorphism, and we can add it if there's ever a use case.

Reply:
I removed it from the `internal/providers` interface and implemented a gRPC error.